.github/workflows/qemu.yaml

```diff
@@ -0,0 +1,72 @@
+name: QEMU environment
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  release:
+    types:
+      - published
+
+jobs:
+  build-linux-arm64:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Install VM dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y qemu-system-arm qemu-efi-aarch64 qemu-utils cloud-image-utils wget
+      - name: Provision VM
+        run: |
+          wget https://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-arm64.img
+
+          qemu-img create -f qcow2 -F qcow2 \
+            -b jammy-server-cloudimg-arm64.img \
+            jammy-snapshot.qcow2 200G
+
+          cat >"user-data" << EOF
+          #cloud-config
+          password: passw0rd
+          chpasswd: { expire: False }
+          ssh_pwauth: True
+          EOF
+          cloud-localds user-data.img user-data
+
+          cp /usr/share/AAVMF/AAVMF_CODE.fd .
+          cp /usr/share/AAVMF/AAVMF_VARS.fd .
+
+          randmac=$(openssl rand -hex 6 | sed 's/\(..\)/\1:/g; s/:$//')
```

Comment:
why are we randomly assigning a MAC?  qemu will auto allocate one for you (it is already well known and we don't really care).